catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java

```diff
@@ -522,6 +535,128 @@ public void init() {
             return "";
         });
 
+        post("/annotations", (req, res) -> {
+            Map<String, Object> incoming = JsonFactory.create()
+                    .parser()
+                    .parseMap(util.safeGetBody(req));
+            String workspaceId = incoming.get("workspace")
+                    .toString();
+            String queryId = incoming.get("parent")
+                    .toString();
+            String annotation = incoming.get("note")
+                    .toString();
+            String user = getSubjectEmail();
+            if (user == null) {
+                return util.getResponseWrapper("error",
+                        "You are not authorized to create notes! A user email is required. "
+                                + "Please ensure you are logged in and/or have a valid email registered in the system.");
+            }
+            if (StringUtils.isBlank(annotation)) {
+                return util.getResponseWrapper("error", "No annotation!");
+            }
+            NoteMetacard noteMetacard = new NoteMetacard(queryId, user, annotation);
+
+            Metacard workspaceMetacard = util.findWorkspace(workspaceId);
+
+            if (workspaceMetacard == null) {
+                return util.getResponseWrapper("error", "Cannot find the workspace metacard!");
+            }
+
+            util.copyAttributes(workspaceMetacard, SECURITY_ATTRIBUTES, noteMetacard);
+
+            Metacard note = saveMetacard(noteMetacard);
+
+            SecurityLogger.auditWarn(
+                    "Attaching an annotation to a resource: resource={} annotation={}",
+                    SecurityUtils.getSubject(),
+                    workspaceId,
+                    noteMetacard.getId());
+
+            Map<String, String> responseNote = noteUtil.getResponseNote(note);
+            if (responseNote == null) {
+                return util.getResponseWrapper("error", "Cannot serialize note metacard to json!");
```

Comment:
❗️ We need to be setting the correct http status codes on the response when we return errors for many of the places in here.

Reply:
ok, i set a code for every error response for annotations. Please review my choice for codes and let me know if there is a better choice for the code.